sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ConfigProperties.java

```diff
@@ -6,131 +6,55 @@
 package io.opentelemetry.sdk.autoconfigure;
 
 import java.time.Duration;
-import java.util.AbstractMap;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
-import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 
-/**
- * Properties to be used for auto-configuration of the OpenTelemetry SDK components. These
- * properties will be a combination of system properties and environment variables. The properties
- * for both of these will be normalized to be all lower case, and underscores will be replaced with
- * periods.
- */
-public final class ConfigProperties {
-
-  private final Map<String, String> config;
-
-  static ConfigProperties get() {
-    return new ConfigProperties(System.getProperties(), System.getenv());
-  }
-
-  // Visible for testing
-  static ConfigProperties createForTest(Map<String, String> properties) {
-    return new ConfigProperties(properties, Collections.emptyMap());
-  }
-
-  private ConfigProperties(Map<?, ?> systemProperties, Map<String, String> environmentVariables) {
-    Map<String, String> config = new HashMap<>();
-    environmentVariables.forEach(
-        (name, value) -> config.put(name.toLowerCase(Locale.ROOT).replace('_', '.'), value));
-    systemProperties.forEach(
-        (key, value) ->
-            config.put(((String) key).toLowerCase(Locale.ROOT).replace('-', '.'), (String) value));
-
-    this.config = config;
-  }
+/** Properties used for auto-configuration of the OpenTelemetry SDK components. */
+public interface ConfigProperties {
 
   /**
    * Returns a string-valued configuration property.
    *
    * @return null if the property has not been configured.
    */
   @Nullable
-  public String getString(String name) {
-    return config.get(name);
-  }
+  String getString(String name);
+
+  /**
+   * Returns a boolean-valued configuration property. Uses the same rules as {@link
+   * Boolean#parseBoolean(String)} for handling the values.
```

Comment:
"Uses the same rules as {@link Boolean#parseBoolean(String)} for handling the values." is an implementation detail.

Reply:
I don't agree. I think this is just shorthand for "read those other javadocs to see how this is parsed". The other option would be to copy the contents of that javadoc over here.

Reply:
Oh, you're saying this probably doesn't belong on the interface javadoc. Hmm. Perhaps changing it to "Implementations should use the same rules as ..." would be the right thing to do here.